plugins/pci-psp/fu-plugin-pci-psp.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (C) 2022 Advanced Micro Devices Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+struct FuPluginData {
+	gchar *sysfs_path;
+};
+
+static void
+fu_plugin_pci_psp_init(FuPlugin *plugin)
+{
+	fu_plugin_alloc_data(plugin, sizeof(FuPluginData));
+	fu_plugin_add_udev_subsystem(plugin, "pci");
+}
+
+static void
+fu_plugin_pci_psp_destroy(FuPlugin *plugin)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+	g_free(priv->sysfs_path);
+}
+
+static gboolean
+fu_plugin_pci_psp_backend_device_added(FuPlugin *plugin, FuDevice *device, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+	g_autofree gchar *fn = NULL;
+
+	/* interesting device? */
+	if (!FU_IS_UDEV_DEVICE(device))
+		return TRUE;
+	if (g_strcmp0(fu_udev_device_get_subsystem(FU_UDEV_DEVICE(device)), "pci") != 0)
+		return TRUE;
+
+	priv->sysfs_path = g_strdup(fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device)));
+	fn = g_build_filename(priv->sysfs_path, "tsme_status", NULL);
+	if (!g_file_test(fn, G_FILE_TEST_EXISTS)) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "PSP or kernel does not export attributes to %s",
+			    priv->sysfs_path);
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_pci_psp_get_attr(FuPlugin *plugin,
+			   FwupdSecurityAttr *attr,
+			   const gchar *file,
+			   gchar *out,
```

Comment:
it feels a bit messy to pass back a char of an int. Maybe do `fu_common_strtoull()`?

Reply:
the kernel only exports 0/1.  Maybe just pass back a bool.

Reply:
WFM.